main.py

```diff
@@ -196,10 +200,36 @@ def run_main():
 elif len(sys.argv) == 1:
     # the name of the executed file always gets passed
     run_main()
+
+elif len(sys.argv) == 4 and sys.argv[1] == '--new-token' and sys.argv[2] == '--file' and sys.argv[3]:
+    with open(sys.argv[3]) as json_file:
```

Comment:
How about encapsulating this into a separate file?
This file is intended to coordinate the general flow.

Reply:
I think the effort of refactoring this into a seperate file is not worth the effort, as this behaviour will only be executed when specifically initiated and the reuse scope would be very narrow. 
Unless you really want this change, I'd drop it.